Tools/Scripts/webkitpy/style/checkers/jsonchecker.py

```diff
@@ -418,3 +420,108 @@ def check_codegen_properties(self, property_name, codegen_properties):
+
+        for key, value in expectations.items():
+            valid = True
+
+            if not isinstance(key, str):
+                self._handle_style_error(
+                    0,
+                    "json/syntax",
+                    5,
+                    "Each key must be a string; {!r} is a {}".format(key, type(key)),
+                )
+                valid = False
+            elif not key.startswith("web-platform-tests/"):
+                self._handle_style_error(
+                    0,
+                    "json/syntax",
+                    5,
+                    "Each key must start with 'web-platform-tests/', got {!r}".format(
+                        key
+                    ),
+                )
+                valid = False
+
+            if value not in ("import", "skip"):
+                self._handle_style_error(
+                    0, "json/syntax", 5, 'Each value must be either "import" or "skip"'
+                )
+                valid = False
+
+            if valid:
+                parsed_expectations[tuple(key.split("/"))] = value
+
+        for parsed_key, value in parsed_expectations.items():
+            if not parsed_key[-1]:
+                self._handle_style_error(
+                    0,
+                    "json/syntax",
+                    5,
+                    "{!r} has trailing slash".format("/".join(parsed_key)),
+                )
+
+            for i in range(len(parsed_key) - 1, 0, -1):
+                parent_key = parsed_key[:i]
+                if parent_key in parsed_expectations:
+                    if value == parsed_expectations[parent_key]:
+                        self._handle_style_error(
+                            0,
+                            "json/syntax",
+                            5,
+                            "{!r} is redundant, {!r} already defines {!r}".format(
+                                "/".join(parsed_key), "/".join(parent_key), value
+                            ),
+                        )
+                    break
+
+            # web-platform-tests/css should be magic, and treated the same as the
+            # top-level; c.f. https://bugs.webkit.org/show_bug.cgi?id=214006
+            if len(parsed_key) == 1 or parsed_key == ("web-platform-tests", "css"):
+                continue
+
+            is_skipped = value == "skip"
+
+            full_path = os.path.join(
+                os.path.dirname(self._file_path), "..", *parsed_key
+            )
+            exists = os.path.exists(full_path)
+            if exists == is_skipped:
+                self._handle_style_error(
+                    0,
+                    "json/syntax",
+                    5,
+                    "{!r} does {}exist and is {}skipped".format(
+                        "/".join(parsed_key),
+                        "" if exists else "not ",
+                        "" if is_skipped else "not ",
+                    ),
+                )
```

Comment:
I think this is actually wrong, given:

```json
{
    "web-platform-tests/a": "skip",
    "web-platform-tests/a/b": "import"
}
```

…should have `a` existing despite being skipped. I guess we need to check if we have any expectations where the current path is a prefix.